generator/graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/generator/SchemaGenerator.kt

```diff
@@ -72,26 +74,36 @@ open class SchemaGenerator(internal val config: SchemaGeneratorConfig) : Closeab
         mutations: List<TopLevelObject> = emptyList(),
         subscriptions: List<TopLevelObject> = emptyList(),
         additionalTypes: Set<KType> = emptySet(),
-        additionalInputTypes: Set<KType> = emptySet()
+        additionalInputTypes: Set<KType> = emptySet(),
+        schemaObject: TopLevelObject? = null
     ): GraphQLSchema {
 
         this.additionalTypes.addAll(additionalTypes.map { AdditionalType(it, inputType = false) })
         this.additionalTypes.addAll(additionalInputTypes.map { AdditionalType(it, inputType = true) })
 
-        val builder = GraphQLSchema.newSchema()
-        builder.query(generateQueries(this, queries))
-        builder.mutation(generateMutations(this, mutations))
-        builder.subscription(generateSubscriptions(this, subscriptions))
-        builder.additionalTypes(generateAdditionalTypes())
-        builder.additionalDirectives(directives.values.toSet())
-
         if (!config.introspectionEnabled) {
             codeRegistry.fieldVisibility(NoIntrospectionGraphqlFieldVisibility.NO_INTROSPECTION_FIELD_VISIBILITY)
         }
 
-        builder.codeRegistry(codeRegistry.build())
+        return GraphQLSchema.newSchema()
+            .query(generateQueries(this, queries))
+            .mutation(generateMutations(this, mutations))
+            .subscription(generateSubscriptions(this, subscriptions))
+            .additionalTypes(generateAdditionalTypes())
+            .additionalDirectives(directives.values.toSet())
+            .codeRegistry(codeRegistry.build())
+            .also { builder ->
+                if (schemaObject != null) {
+                    directives.clear()
 
-        return config.hooks.willBuildSchema(builder).build()
+                    builder.description(schemaObject.kClass.getGraphQLDescription())
+                        .withSchemaAppliedDirectives(generateSchemaDirectives(this, schemaObject))
+                        .withSchemaDirectives(directives.values)
+                }
+            }
+            .let { builder ->
```

Comment:
nip: should we move hook invocation to the `.also` scope function ? usually `let` is used to execute lambdas on non-null objects
also let returns lambda result, this suit more as a side effect, and we would be returning result of hook.